connector/connectortest/fanoutconsumer.go

```diff
@@ -0,0 +1,108 @@
+// Copyright The OpenTelemetry Authors
```

Comment:
I suggest we call this file `router.go`. Technically it wraps fanoutoutconsumer, but we're still only exposing it as a "router"